dot/network/transaction.go

```diff
@@ -191,6 +193,10 @@ func decodeTransactionMessage(in []byte) (NotificationsMessage, error) {
 }
 
 func (s *Service) handleTransactionMessage(peerID peer.ID, msg NotificationsMessage) (bool, error) {
+	if msg.IsHandshake() {
+		return false, errExpectedTransactionMsg
+	}
```

Comment:
A `*TransactionMessage` always returns false to `IsHandshake`, so this check is unneeded given the below type assertion.

Also why not having `msg *TransactionMessage` instead of using an interface `msg NotificationsMessage` for the argument? Especially given the function name, type check/switching should be done at the caller 🤔

Reply:
same reply as https://github.com/ChainSafe/gossamer/pull/2681#discussion_r925750884
handleTransactionMessage is similar to handleBlockAnnounceMessage

Reply:
I still don't understand.

`TransactionMessage` always returns false for `IsHandshake()`, so the type assertion below is fine.
Unless you are doing something with the error returned, such as `errors.Is()`, but I don't think that's the case right?

Reply:
addressed